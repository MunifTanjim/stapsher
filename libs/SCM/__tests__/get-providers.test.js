const helpers = require('../../../__tests__/helpers')

helpers.addSnapshotSerializers()

const serverConfig = require('../../../configs/server')

const GitHub = require('../GitHub')
const GitLab = require('../GitLab')

const getProviders = require('../get-providers')

describe('libs/SCM/get-providers', () => {
  it('returns object containing provider constructor functions', () => {
    serverConfig.set('scmProviders', ['github.bot', 'gitlab.bot'])

    let providers = getProviders()

    expect(providers).toHaveProperty('github')
    expect(providers).toHaveProperty('gitlab')
    expect(providers.github()).toBeInstanceOf(GitHub)
    expect(providers.gitlab()).toBeInstanceOf(GitLab)
  })

  it('throws error if invalid config', () => {
    serverConfig.set('scmProviders', ['github.bot'])
    serverConfig.set('github.bot.accessToken', '')

    try {
      getProviders()
    } catch (err) {
      expect(err).toMatchSnapshot()
    }
  })
})
