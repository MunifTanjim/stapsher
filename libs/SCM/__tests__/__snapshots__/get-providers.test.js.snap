// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`libs/SCM/get-providers throws error if invalid config 1`] = `"Error: Invalid config for SCM provider: github.bot"`;
